### ISSUE TITLE
libepoxy build failed

### ISSUE THREAD

**Issue by koir1100** [wingtk/gvsbuild#488]
Hello I struggle for build gtk3 into gvsbuild, but libepoxy step build failed...

I don't know how fix it. So please help me...
Build environment is msvc 2017 32bit, python 3.10.2.
I'll show the log.

> python .\build.py build -p x86 --vs-ver 15 --python-dir "C:\Python310-32" --same-python --clean --gtk3-ver 3.24 libepoxy
> Cleaning up the build environment
> Checking msys tool
> Checking Msvc tool
> Downloading packages
> Building project nuget (5.4.0)
> Building project python ()
> Building project ninja (1.8.2)
> Building project meson (0.57.1)
> (tar) Exporting libepoxy
> Building project libepoxy (1.5.5)
> The Meson build system
> Version: 0.57.1
> Source dir: C:\gtk-build\build\Win32\release\libepoxy
> Build dir: C:\gtk-build\build\Win32\release\libepoxy\_gvsbuild-meson
> Build type: native build
> Project name: libepoxy
> Project version: 1.5.5
> C compiler for the host machine: cl (msvc 19.16.27045 "Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27045 for x86")
> C linker for the host machine: link link 14.16.27045.0
> Host machine cpu family: x86
> Host machine cpu: x86
> cl.exe does not actually support gnu standards, and meson will instead demote to the nearest ISO C standard. This may cause compilation to fail.
> Has header "KHR/khrplatform.h" : NO
> Compiler for C supports arguments -we4002: YES
> Compiler for C supports arguments -we4003: YES
> Compiler for C supports arguments -w14010: YES
> Compiler for C supports arguments -we4013: YES
> Compiler for C supports arguments -w14016: YES
> Compiler for C supports arguments -we4020: YES
> Compiler for C supports arguments -we4021: YES
> Compiler for C supports arguments -we4027: YES
> Compiler for C supports arguments -we4029: YES
> Compiler for C supports arguments -we4033: YES
> Compiler for C supports arguments -we4035: YES
> Compiler for C supports arguments -we4045: YES
> Compiler for C supports arguments -we4047: YES
> Compiler for C supports arguments -we4049: YES
> Compiler for C supports arguments -we4053: YES
> Compiler for C supports arguments -we4071: YES
> Compiler for C supports arguments -we4819: YES
> Library dl found: NO
> Found pkg-config: C:\gtk-build\gtk\Win32\release\bin\pkg-config.EXE (1.8.0)
> Run-time dependency gl found: YES
> Found CMake: C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.EXE (3.12.18081601)
> Run-time dependency egl found: NO (tried pkgconfig and cmake)
> Run-time dependency x11 found: NO (tried pkgconfig and cmake)
> Run-time dependency glesv2 found: NO (tried pkgconfig and cmake)
> Library libGLESv2 found: NO
> Run-time dependency glesv1_cm found: NO (tried pkgconfig and cmake)
> Library libGLESv1_CM found: NO
> Library opengl32 found: YES
> Library gdi32 found: YES
> Program src/gen_dispatch.py found: YES (C:\Python310-32\python.exe C:\gtk-build\build\Win32\release\libepoxy\src/gen_dispatch.py)
> Configuring config.h using configuration
> Library dl found: NO
> Checking for function "dlvsym" with dependency -ldl: NO
> Build targets in project: 14
> 
> Found ninja-1.8.2 at C:\gtk-build\tools\ninja-1.8.2\ninja.EXE
> [5/30] Compiling C object "test/khronos typedefs.exe.p/khronos_typedefs_nonepoxy.c.obj"
> ../test/khronos_typedefs_nonepoxy.c(60): warning C4068: unknown pragma
> [10/30] Compiling C object src/epoxy-0.dll.p/dispatch_common.c.obj
> FAILED: src/epoxy-0.dll.p/dispatch_common.c.obj
> "cl" "-Isrc\epoxy-0.dll.p" "-Isrc" "-I..\src" "-Iinclude" "-I..\include" "-Iinclude\epoxy" "/MD" "/nologo" "/showIncludes" "/W2" "/O2" "/Zi" "-we4002" "-we4003" "-w14010" "-we4013" "-w14016" "-we4020" "-we4021" "-we4027" "-we4029" "-we4033" "-we4035" "-we4045" "-we4047" "-we4049" "-we4053" "-we4071" "-we4819" "/Fdsrc\epoxy-0.dll.p\dispatch_common.c.pdb" /Fosrc/epoxy-0.dll.p/dispatch_common.c.obj "/c" ../src/dispatch_common.c
> ../src/dispatch_common.c: error C4819: The file contains a character that cannot be represented in the current code page (949). Save the file in Unicode format to prevent data loss
> [11/30] Compiling C object src/epoxy-0.dll.p/dispatch_wgl.c.obj
> C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\minwindef.h(130): warning C4005: 'APIENTRY': macro redefinition
> C:\gtk-build\build\Win32\release\libepoxy\include\epoxy/gl.h(59): note: see previous definition of 'APIENTRY'
> [13/30] Compiling C object src/epoxy-0.dll.p/meson-generated_.._wgl_generated_dispatch.c.obj
> C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\minwindef.h(130): warning C4005: 'APIENTRY': macro redefinition
> C:\gtk-build\build\Win32\release\libepoxy\include\epoxy/gl.h(59): note: see previous definition of 'APIENTRY'
> src/wgl_generated_dispatch.c(4557): warning C4098: 'epoxy_wglBlitContextFramebufferAMD_global_rewrite_ptr': 'void' function returning a value
> src/wgl_generated_dispatch.c(4557): warning C4098: 'epoxy_wglBlitContextFramebufferAMD_dispatch_table_rewrite_ptr': 'void' function returning a value
> src/wgl_generated_dispatch.c(4557): warning C4098: 'epoxy_wglBlitContextFramebufferAMD_dispatch_table_thunk': 'void' function returning a value
> src/wgl_generated_dispatch.c(4583): warning C4098: 'epoxy_wglDeleteBufferRegionARB_global_rewrite_ptr': 'void' function returning a value
> src/wgl_generated_dispatch.c(4583): warning C4098: 'epoxy_wglDeleteBufferRegionARB_dispatch_table_rewrite_ptr': 'void' function returning a value
> src/wgl_generated_dispatch.c(4583): warning C4098: 'epoxy_wglDeleteBufferRegionARB_dispatch_table_thunk': 'void' function returning a value
> src/wgl_generated_dispatch.c(4587): warning C4098: 'epoxy_wglDestroyDisplayColorTableEXT_global_rewrite_ptr': 'void' function returning a value
> src/wgl_generated_dispatch.c(4587): warning C4098: 'epoxy_wglDestroyDisplayColorTableEXT_dispatch_table_rewrite_ptr': 'void' function returning a value
> src/wgl_generated_dispatch.c(4587): warning C4098: 'epoxy_wglDestroyDisplayColorTableEXT_dispatch_table_thunk': 'void' function returning a value
> [18/30] Compiling C object src/epoxy-0.dll.p/meson-generated_.._gl_generated_dispatch.c.obj
> C:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared\minwindef.h(130): warning C4005: 'APIENTRY': macro redefinition
> C:\gtk-build\build\Win32\release\libepoxy\include\epoxy/gl.h(59): note: see previous definition of 'APIENTRY'
> ninja: build stopped: subcommand failed.
> Traceback (most recent call last):
>   File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 589, in build
>     if self.__build_one(p):
>   File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 743, in __build_one
>     skip_deps = proj.build()
>   File "C:\gtk-build\github\gvsbuild\gvsbuild\projects.py", line 1266, in build
>     Meson.build(self)
>   File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\base_builders.py", line 88, in build
>     self.builder.exec_ninja(params="install", working_dir=ninja_build)
>   File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 1091, in exec_ninja
>     self.__execute(
>   File "C:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 1152, in __execute
>     subprocess.check_call(args, cwd=working_dir, env=env, shell=True)
>   File "C:\Python310-32\lib\subprocess.py", line 369, in check_call
>     raise CalledProcessError(retcode, cmd)
> subprocess.CalledProcessError: Command 'ninja install' returned non-zero exit status 1.
> Error: libepoxy build failed

Thank you!!!

**Comment by koir1100** [comment]
I found a solution.

[This page](https://blog.naver.com/jkyun93/222044324510) said to remove all annotations from both dispatch_common.c and dispatch_common.h files to avoid errors.

In this case, These files will exist in the following two folders respectively.
A: C:\gtk-build\build\Win32\release\libepoxy\src, B: C:\gtk-build\build\Win32\release\gtk3-24\subprojects\libepoxy\src

I tried, not occur the error. But I think that this solution may be a stopgap action...
To solve the problem, Maybe find another fundamental solution definitely. Thank you.
